File: Programmers/week4/76502/jjunii/76502.py
from collections import deque

def is_correct(str):
    pairs = {')': '(', ']': '[', '}': '{'}
    
    stack = []
    for c in str:
        if c in ['(', '[', '{']:
            stack.append(c)
        elif stack and stack[-1] == pairs[c]:
            stack.pop()
        else:
            return False
        
    return False if stack else True

def solution(s):
    answer = 0
    s = deque(s)
    for i in range(len(s)):
        s.rotate(-1)
        
        if s[0] not in ['(', '[', '{']:
            continue
            
        if is_correct(s):
             answer += 1
    return answer

File: Programmers/week4/76502/kwonjin2/76502.js
function solution(s) {
    let count = 0;
    
    for (let i = 0; i < s.length; i++) {
        if (isValid(s)) count++;
        s = s.slice(1) + s[0];
    }
    
    return count;
}

const isValid = (s) => {
    const stack = [];
    const pairs = { '(': ')', '[': ']', '{': '}' };
    
    for (let c of s) {
        if (c === '(' || c === '[' || c === '{') {
            stack.push(c);
        } else {
            if (pairs[stack.pop()] !== c) return false;
        }
    }
    
    return stack.length === 0;
}
